packages/expo-image-picker/android/src/main/java/expo/modules/imagepicker/crop/ImageLibraryContract.kt

```diff
@@ -0,0 +1,60 @@
+package expo.modules.imagepicker.crop
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import androidx.activity.result.contract.ActivityResultContract
+import expo.modules.imagepicker.MediaType
+
+
+/**
+ * An [ActivityResultContract] to prompt the user to pick single or multiple image(s) or/and video(s),
+ * receiving a `content://` [Uri] for each piece of content.
+ *
+ * @see [ActivityResultContracts.GetContent] or [ActivityResultContracts.GetMultipleContents]
+ */
+internal class ImageLibraryContract(
+  private val allowMultipleSelection: Boolean,
+  private val singleMimeType: String,
+  private val multipleMimeTypes: Array<String>?
+): MediaPickerContract() {
+  override fun createIntent(context: Context, input: Any?) =
+    Intent(Intent.ACTION_GET_CONTENT)
+      .addCategory(Intent.CATEGORY_OPENABLE)
+      .setType(singleMimeType)
+      .apply {
+        if (allowMultipleSelection) {
+          putExtra(Intent.EXTRA_ALLOW_MULTIPLE, true)
+          putExtra(Intent.EXTRA_MIME_TYPES, multipleMimeTypes)
+        }
+      }
+
+  override fun parseResult(resultCode: Int, intent: Intent?) = MediaPickerContractResult(
+    cancelled = resultCode == Activity.RESULT_CANCELED,
+    data = intent.takeIf { resultCode == Activity.RESULT_OK }?.getClipDataUris() ?: emptyList()
+  )
+
+  internal companion object {
+    internal fun Intent.getClipDataUris(): List<Pair<MediaType, Uri>> {
```

Comment:
- I'd extract this function from companion object to global file scope
- Isn't `List<Pair<X, Y>>` just a `Map<X, Y>` 😅 If you need list of pairs, any map has a `.toList()` function for that